src/output.cpp

```diff
@@ -2011,99 +2012,106 @@ static void do_kw_subst(chunk_t *pc)
 } // do_kw_subst
 
 
-static void output_comment_multi_simple(chunk_t *pc, bool kw_subst)
+static void output_comment_multi_simple(chunk_t *pc)
 {
-   UNUSED(kw_subst);
+   if (pc == nullptr)
+   {
+      return;
+   }
+
    cmt_reflow cmt;
    output_cmt_start(cmt, pc);
 
-   int col_diff = 0;
-   if (chunk_is_newline(chunk_get_prev(pc)))
-   {
-      // The comment should be indented correctly
-      col_diff = pc->orig_col - pc->column;
-   }
-   else
-   {
-      // The comment starts after something else
-      col_diff = 0;
-   }
+   // The multiline comment is saved inside one chunk. If the comment is
+   // shifted all lines of the comment need to be shifter by the same amount.
+   // Save the difference of initial and current position to apply it on every
+   // line_column
+   const int col_diff = [pc](){
+      int diff = 0;
+
+      if (chunk_is_newline(chunk_get_prev(pc)))
+      {
+         // The comment should be indented correctly
+         diff = pc->column - pc->orig_col;
+      }
+
+      return(diff);
+   } ();
 
    unc_text line;
-   line.clear();
-   size_t   line_count = 0;
-   size_t   cmt_idx    = 0;
-   size_t   ccol       = pc->column;
-   bool     nl_end     = false;
+   size_t   line_count  = 0;
+   size_t   line_column = pc->column;
+   size_t   cmt_idx     = 0;
    while (cmt_idx < pc->len())
    {
-      int ch = pc->str[cmt_idx++];
-
-      // handle the CRLF and CR endings. convert both to LF
-      if (ch == '\r')
-      {
-         ch = '\n';
-         if ((cmt_idx < pc->len()) && (pc->str[cmt_idx] == '\n'))
-         {
-            cmt_idx++;
-         }
-      }
+      int ch = pc->str[cmt_idx];
+      cmt_idx++;
 
-      // Find the start column
+      // 1: step through leading tabs and spaces to find the start column
       if (line.size() == 0)
       {
-         nl_end = false;
          if (ch == ' ')
          {
-            ccol++;
+            line_column++;
             continue;
          }
          else if (ch == '\t')
          {
-            ccol = calc_next_tab_column(ccol, cpd.settings[UO_input_tab_size].u);
+            line_column = calc_next_tab_column(line_column, cpd.settings[UO_input_tab_size].u);
             continue;
          }
-         else
+      }
+
+      // 2: add chars to line, handle the CRLF and CR endings (convert both to LF)
```

Comment:
> convert both to LF

why? isn't that controlled by an option?

Reply:
I just refactored the function to make it more understandable (mostly for my own sake).  While doing that I didn't gave too much thought into actually changing its behaviour outside of what I needed to change to fix the reported issue. 

Here the `\r\n` are converted into `\n`, this most certainly isn't needed here. 
The actual determination which line ends are going to be used in the end are happening inside the end of the tokenization function (using the `UO_newlines` option) and the write char function which actually is writing the correct line ends.

Reply:
Dropping the check will force me to make nasty checks in other places. 
Why do we keep the \r around anyways? I will try to remove them in the tokenization and see if that causes a total mess or simplifies things even more.